vendor/plugins/edu.harvard.dce.paella.timedCommentsPlugin/timedCommentsOverlay.js

```diff
@@ -123,24 +127,142 @@ Class ("paella.plugins.TimedCommentsOverlay", paella.EventDrivenPlugin, {
     thisClass.lastEvent = eventType;
   },
   
+  // This gets the user access roles from Opencast user service
+  // Required to determin if user is logged in as admin.
   getUserData: function () {
     var self = this;
     var defer = new $.Deferred();
-    if (self._userData) {
-      defer.resolve(self._userData);
-    } else {
-      paella.opencast.getUserInfo().then(
-      function (me) {
-        self._userData = me;
+    // always refresh userdata
+    paella.opencast.getUserInfo().then(
+    function (me) {
+      self._userData = me;
+      // If not loggged in as admin, do the pseudo name check
+      if ($.inArray('ROLE_ADMIN', me.roles) === -1) {
+        self.getPseudoName().then(function (pseudoName) {
+          // replacing OC username with annot pseudo name and
+          // also setting flag that its an annot pseudoname
+          self._userData.username = pseudoName;
+          self._userData.pseudoName = pseudoName;
+          defer.resolve(self._userData);
+        });
+      } else {
         defer.resolve(self._userData);
-      },
-      function () {
-        defer.reject();
-      });
-    }
+      }
+    },
+    function () {
+      defer.reject();
+    });
+    return defer;
+  },
+  
+  // This querires the annot DCE endpoint for the users annot username
+  // WARN: This query is only useful when the user is NOT logged in with Admin roles.
+  getPseudoName: function () {
+    var thisClass = this;
+    var defer = new $.Deferred();
+    paella.data.read('timedComments', {
+      id: paella.initDelegate.getId(),
+      question: 'getMyPseudoName'
+    },
+    function (data) {
+      base.log.debug("TC user's pseuduo name " + data);
+      thisClass._currentPseudoName = data;
+      defer.resolve(data);
+    },
+    function () {
+      base.log.debug("TC ERROR getting user's pseuduo name -1.");
+      defer.reject();
+    });
+    return defer;
+  },
+  
+  setPseudoName: function (newPseudoName, pseudoNameInputElem) {
+    var thisClass = this;
+    var defer = new $.Deferred();
+    base.log.debug("TC about to set pseuduo name " + newPseudoName);
+    paella.data.write('timedComments', {
+      id: paella.initDelegate.getId(),
+      newPseudoName: newPseudoName
+    },
+    "fillervalue",
+    function (data, returnCode, isSuccess) {
+      if (returnCode === 400 || returnCode === 409) {
+        $(pseudoNameInputElem).val("");
+        $(pseudoNameInputElem).attr('placeholder', 'Alias "' + newPseudoName + '" is already taken, try again or leave blank');
+        defer.reject(returnCode);
+      } else if (isSuccess) {
+        base.log.debug("TC user's pseuduo name " + data);
+        thisClass._currentPseudoName = data;
+        defer.resolve(data);
+      } else {
+        $(pseudoNameInputElem).attr('placeholder', 'Unable to set "' + newPseudoName + '" at the moment. Try again later.');
```

Comment:
What happens to get us here? You mentioned the pseudo name string can be only ascii. If I try to use some higher unicode character it will be rejected?